treq/test/test_auth.py

```diff
@@ -42,6 +48,61 @@ def test_overrides_per_request_headers(self):
         )
 
 
+class PinToFirstHostAgentTests(TestCase):
+    def setUp(self):
+        self.first_agent = mock.Mock(Agent)
```

Comment:
I may like `mock` less than any other person on the planet.  I chose it here to stay consistent with the existing test case, since a person wandering in would wonder why one used `mock` and the others `treq.testing`.  I'd be happy to do a follow up PR that switched things over to `treq.testing`.